barretenberg/cpp/src/barretenberg/honk/flavor/goblin_translator.hpp

```diff
@@ -0,0 +1,1630 @@
+        QUOTIENT_LOW_BINARY_LIMBS_SHIFT,
+        QUOTIENT_HIGH_BINARY_LIMBS_SHIFT,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAINT_0_SHIFT,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAINT_1_SHIFT,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAINT_2_SHIFT,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAINT_3_SHIFT,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAINT_4_SHIFT,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAINT_TAIL_SHIFT,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAINT_0_SHIFT,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAINT_1_SHIFT,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAINT_2_SHIFT,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAINT_3_SHIFT,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAINT_4_SHIFT,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAINT_TAIL_SHIFT,
+        RELATION_WIDE_LIMBS_SHIFT,
+        RELATION_WIDE_LIMBS_RANGE_CONSTRAINT_0_SHIFT,
+        RELATION_WIDE_LIMBS_RANGE_CONSTRAINT_1_SHIFT,
+        RELATION_WIDE_LIMBS_RANGE_CONSTRAINT_2_SHIFT,
+        RELATION_WIDE_LIMBS_RANGE_CONSTRAINT_3_SHIFT,
+        ORDERED_RANGE_CONSTRAINTS_0_SHIFT,
+        ORDERED_RANGE_CONSTRAINTS_1_SHIFT,
+        ORDERED_RANGE_CONSTRAINTS_2_SHIFT,
+        ORDERED_RANGE_CONSTRAINTS_3_SHIFT,
+        ORDERED_RANGE_CONSTRAINTS_4_SHIFT,
+
+        Z_PERM_SHIFT,
+        /*All precomputed polynomials*/
+        LAGRANGE_FIRST,
+        LAGRANGE_LAST,
+        LAGRANGE_ODD,
+        LAGRANGE_EVEN,
+        LAGRANGE_SECOND,
+        LAGRANGE_SECOND_TO_LAST_IN_MINICIRCUIT,
+        ORDERED_EXTRA_RANGE_CONSTRAINTS_NUMERATOR,
+        /*Utility value*/
+        TOTAL_COUNT
+
+    };
+
+    using CircuitBuilder = GoblinTranslatorCircuitBuilder;
+    using PCS = pcs::kzg::KZG<curve::BN254>;
+    using Curve = curve::BN254;
+    using GroupElement = Curve::Element;
+    using Commitment = Curve::AffineElement;
+    using CommitmentHandle = Curve::AffineElement;
+    using CommitmentKey = pcs::CommitmentKey<Curve>;
+    using VerifierCommitmentKey = pcs::VerifierCommitmentKey<Curve>;
+    using FF = Curve::ScalarField;
+    using BF = Curve::BaseField;
+    using Polynomial = barretenberg::Polynomial<FF>;
+    using PolynomialHandle = std::span<FF>;
+
+    // The size of the circuit which is filled with non-zero values for most polynomials. Most relations (everything
+    // except for Permutation and GenPermSort) can be evaluated just on the first chunk
+    static constexpr size_t MINI_CIRCUIT_SIZE = mini_circuit_size;
+
+    // How many mini_circuit_size polynomials are concatenated in one concatenated_*
+    static constexpr size_t CONCATENATION_INDEX = 16;
+
+    // The number of concatenated_* wires
+    static constexpr size_t NUM_CONCATENATED_WIRES = 4;
+
+    // Actual circuit size
+    static constexpr size_t FULL_CIRCUIT_SIZE = MINI_CIRCUIT_SIZE * CONCATENATION_INDEX;
+
+    // Number of wires
+    static constexpr size_t NUM_WIRES = CircuitBuilder::NUM_WIRES;
+
+    // The step in the GenPermSort relation
+    static constexpr size_t SORT_STEP = 3;
```

Comment:
For these hardcoded params, it would be useful to add a comment explicitly stating that these cannot be changed without breaking the circuit (I'm assuming this is the case)

Reply:
Added